File: Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.Text.RegularExpressions;

namespace Product_Management_System.Entities
{
	public class Product
	{
		public int Id { get; set; }

		public string GeneratedCode { get; set; }

		public string Name { get; set; }

		public string Unit { get; set; }

		public decimal Price { get; set; }

		public int InitialQuantity { get; set; }

		public ICollection<ProductTransaction> Transactions { get; set; }
	}
}

File: wwwroot/js/product.js
$("#productForm").submit(function (e) {
    e.preventDefault();

    var form = $("#productForm");

    if (!form.valid()) {
        return;
    }

    var formData = {
        name: $("input[name='Name']").val(),
        unit: $("input[name='Unit']").val(),
        price: parseFloat($("input[name='Price']").val()),
        initialQuantity: parseInt($("input[name='InitialQuantity']").val()),
        generatedCode: $("input[name='generatedCode']").val()
    };

    $.ajax({
        type: "POST",
        url: "/Product/Create",
        contentType: "application/json",
        data: JSON.stringify(formData),
        success: function (response) {
            if (response.success) {
                $("#message").html(`<div class="alert alert-success">${response.message}</div>`);
                form[0].reset();
                form.find(".text-danger").empty();
                $.get("/Product/GenerateCode", function (newCode) {
                    $("#GeneratedCode").val(newCode);
                });
            } else {
                $("#message").html(`<div class="alert alert-danger">${response.message}</div>`);
            }
        },
        error: function () {
            $("#message").html(`<div class="alert alert-danger">Unexpected error occurred.</div>`);
        }
    });
});
